src/replit/database/database.py

```diff
@@ -174,21 +186,27 @@ async def delete(self, key: str) -> None:
 
     async def list(self, prefix: str) -> Tuple[str, ...]:
         """List keys in the database which start with prefix.
-
         Args:
             prefix (str): The prefix keys must start with, blank not not check.
-
         Returns:
             Tuple[str]: The keys found.
         """
-        params = {"prefix": prefix, "encode": "true"}
-        async with self.sess.get(self.db_url, params=params) as response:
-            response.raise_for_status()
-            text = await response.text()
-            if not text:
-                return tuple()
-            else:
-                return tuple(urllib.parse.unquote(k) for k in text.split("\n"))
+        for i in range(self.retry_count):
```

Comment:
here it's probably better to have a [decorator](https://realpython.com/primer-on-python-decorators/) that does the retries.

or even better, we can use a third-party retry client that already does all of this for us: https://pypi.org/project/aiohttp-retry/